src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -806,8 +806,40 @@ parseMPToken(Json::Value const& mptJson, Json::Value& jvResult)
     }
 }
 
+static std::optional<uint256>
+parsePermissionedDomains(Json::Value const& pd, Json::Value& jvResult)
+{
+    if (pd.isString())
+    {
+        uint256 uNodeIndex;
+        if (uNodeIndex.parseHex(pd.asString()))
+            return uNodeIndex;
+
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+
+    if (!pd.isObject() || !pd.isMember(jss::account) ||
+        !pd[jss::account].isString() || !pd.isMember(jss::seq) ||
+        (pd[jss::seq].isInt() && pd[jss::seq].asInt() < 0) ||
+        (!pd[jss::seq].isInt() && !pd[jss::seq].isUInt()))
```

Comment:
For better error messaging, this should be split up into separate ifs for `account` and `seq`

Reply:
It will return the same `malformedRequest` for both cases, don't see improvement here.

Reply:
Here and below - This will break established rules for LedgerEntry. This require dedicated PR for LedgerEntry improvement

Reply:
`ledger_entry` is very inconsistent with its error codes. You see `malformedAccount` and `malformedSeq` in some of the returns (such as in Escrow processing).

Reply:
Yes, most of the places return `malformedRequest` and a few of them returning specific code. That's why it require additional PR to fix them all.

Reply:
Yes, but I'd rather do it the correct way here.

Reply:
Correct way require these small fixes for full LedgerEntry code, so it is better to do in dedicated PR.